eclair-core/src/main/scala/fr/acinq/eclair/db/psql/PsqlNetworkDb.scala

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2019 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.db.psql
+
+import fr.acinq.bitcoin.{ByteVector32, Crypto, Satoshi}
+import fr.acinq.eclair.ShortChannelId
+import fr.acinq.eclair.db.NetworkDb
+import fr.acinq.eclair.router.Router.PublicChannel
+import fr.acinq.eclair.wire.LightningMessageCodecs.{channelAnnouncementCodec, channelUpdateCodec, nodeAnnouncementCodec}
+import fr.acinq.eclair.wire.{ChannelAnnouncement, ChannelUpdate, NodeAnnouncement}
+import grizzled.slf4j.Logging
+import javax.sql.DataSource
+
+import scala.collection.immutable.SortedMap
+
+class PsqlNetworkDb(implicit ds: DataSource) extends NetworkDb with Logging {
+
+  import PsqlUtils.ExtendedResultSet._
+  import PsqlUtils._
+
+  val DB_NAME = "network"
+  val CURRENT_VERSION = 2
+
+  inTransaction { psql =>
+    using(psql.createStatement()) { statement =>
+      getVersion(statement, DB_NAME, CURRENT_VERSION) match {
+        case CURRENT_VERSION => () // nothing to do
+        case unknown => throw new IllegalArgumentException(s"unknown version $unknown for network db")
+      }
+      statement.executeUpdate("CREATE TABLE IF NOT EXISTS nodes (node_id TEXT NOT NULL PRIMARY KEY, data BYTEA NOT NULL)")
+      statement.executeUpdate("CREATE TABLE IF NOT EXISTS channels (short_channel_id BIGINT NOT NULL PRIMARY KEY, txid TEXT NOT NULL, channel_announcement BYTEA NOT NULL, capacity_sat BIGINT NOT NULL, channel_update_1 BYTEA NULL, channel_update_2 BYTEA NULL)")
```

Comment:
Can `capacity_sat` be an `INTEGER` instead of `BIGINT`?

Reply:
Scala's `Long` doen't fit Postgres' `INTEGER`.